src/qforte/sq_operator.h

```diff
@@ -26,36 +29,39 @@ class SQOperator {
     /// add an second quantized operator to the second quantized operator
     void add_op(const SQOperator& sqo);
 
-    /// sets the operator coefficeints
+    /// sets the operator coefficients
     void set_coeffs(const std::vector<std::complex<double>>& new_coeffs);
 
-    /// multiplies the sq operator coefficeints by multiplier
+    /// multiplies the sq operator coefficients by multiplier
     void mult_coeffs(const std::complex<double>& multiplier);
 
-    /// return a vector of terms and thier coeficients
+    /// return a vector of terms and their coefficients
     const std::vector<std::pair< std::complex<double>, std::vector<size_t>>>& terms() const;
 
-    /// order a single term
+    /// Put a single term into "canonical" form. Canonical form orders orbital indices
+    /// descending.
     void canonical_order_single_term(std::pair< std::complex<double>, std::vector<size_t>>& term );
 
-    /// order each product of ac operators in a standardized fashion
+    /// Canonicalize each term. The order of the terms is unaffected.
     void canonical_order();
 
-    /// simplify the operator (i.e. combine like terms)
+    /// Combine like terms in terms_. As a side-effect, canonicalizes the order.
     void simplify();
 
-    /// return the QuantumOperator ojbect correstponting the the Jordan-Wigner
+    /// return the QuantumOperator object corresponding the the Jordan-Wigner
     /// transform of this sq operator.
     QuantumOperator jw_transform();
 
     /// return a vector of string representing this quantum operator
     std::string str() const;
 
   private:
-    /// the list of circuits
+    /// The linear combination of second quantized operators. Stored in pairs of
+    /// coefficients, and then a vector of N created indices, followed by N annihilated indices.
+    /// Orbital indices start at zero.
     std::vector<std::pair< std::complex<double>, std::vector<size_t>>> terms_;
```

Comment:
For another PR: the assumption that we have N and N creators/annihilators is not necessary. This could be generalized by turning it into a tuple(coefficient, vec(int), vec(int)) or a pair(coefficient,vec(pair(orb,is_creation))).